### ISSUE TITLE
Update models in the demo

### ISSUE THREAD

**Issue by matt-gardner** [allenai/allennlp#3757]
This isn't really an allennlp issue, but I'm putting it here so that we can track it in our 1.0 milestone.  We should update our demo to include the new models that we're building.  Specifically, at least:

- [ ] [TransformerQA](https://github.com/allenai/allennlp-reading-comprehension/pull/19)
- [x] Update coref models
- [x] New DROP NMN model
- [x] NLI models

We could also consider updating some of our "annotate a sentence" models, though those seem like lower-priority items to me.

**Comment by schmmd** [comment]
While it'd be great to have these models, I'm not sure we need to couple this with the 1.0 release--which is more about releasing some major changes we've made to the library.  I worry that--if we tie too many things to the 1.0 release--it won't happen.

**Comment by matt-gardner** [comment]
Note that all of these models are already implemented.  TransformerQA is merged, it just needs to be added to the demo.  The coref model is already in the demo.  The DROP NMN model is implemented, and a new demo for it is in progress.  The NLI models have been implemented satisfactorily in a repo by a UCI student, and they just need to be updated to 1.0 and put into a separate repo, then updated in our demo.  None of what's in this issue is very much work.

**Comment by schmmd** [comment]
Great!

**Comment by matt-gardner** [comment]
Giving this a label of "Day", as what's in this issue isn't very much work, though some of it is blocked on other things that will take longer.

**Comment by schmmd** [comment]
@matt-gardner what's it blocked on?  This is a PR I'm actively working on and not an issue.  @dirkgr already updated most of the models--I just need to work through how to use the hub (and add the right imports) rather than depending on allennlp-core.

**Comment by matt-gardner** [comment]
I saw that other PR and considered linking it, but it's a bit different.  That PR is updating the demo to make sure it works with new code.  What this issue is talking about is more modern models (not just modern code for old models).  In particular, you don't have TransformerQA as an item on that PR, you have BiDAF and ELMo-BiDAF.  It's a separate issue.  Same for new NLI models - what I mean here is not taking our existing model and making sure it works (which is also important), but instead adding a new model based on a transformer (which I believe @ZhaofengWu is working on).  This issue is just about closing the loop after new models are added to the code, to actually put them in the demo.

**Comment by schmmd** [comment]
Sorry--I was confused.

**Comment by ZhaofengWu** [comment]
Yes NLI is on my TODO list, but I'm doing some coref work for now and haven't started working on it. The reason is that we don't seem to have a MNLI reader. I will likely start working on it next week. If we just want an SNLI model, nevertheless, which we already have a reader for, making a config and launching the training will take literally minutes.

**Comment by matt-gardner** [comment]
I know that @anthonywchen has some code for this, though I can't find it now, and it would need a little bit of massaging to get it to use our standard abstractions.  But that's probably better than starting from scratch, as he has models that reproduce published results.

**Comment by anthonywchen** [comment]
Here's [my repository](https://github.com/anthonywchen/generative-nli/) which wraps the HuggingFace models in AllenNLP for finetuning BERT/ROBERTa on MNLI. I've managed to reproduce the published results on MNLI-matched validation set.

As Matt said, I have inconsistencies with my code and how the abstractions in AllenNLP are supposed to be used. E.g. 
- not using Fields as effectively as the could be used
- doing a lot of the data pre-processing out of the DatasetReader class.

Happy to help with getting these models into the demo and let me know if you have any questions.

**Comment by schmmd** [comment]
@ZhaofengWu @matt-gardner are the NLI models done?

![image](https://user-images.githubusercontent.com/954798/81761280-65a00f00-947e-11ea-9dbd-1d11586bcbc3.png)

Is TransformerQA the only model left?  We can now easily make progress on this item, since we've moved to the new demo architecture.

**Comment by ZhaofengWu** [comment]
The NLI models are done in the sense that I added these new models and they were working at the time when I left. If they still work, then I think it's safe to say they're done.

**Comment by epwalsh** [comment]
I'm not seeing an archived Transformer QA model anywhere. Do we not have a trained model yet?
